### PR TITLE
Node notes cli

### PR DESCRIPTION
**Pull request by OSAdams**
node-notes-cli

### CODE REVIEW

File: node-notes-cli/app.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          console.log(`Note ID ${notesObjId} successfully updated`);
+        }
+      }
+      fs.writeFile('data.json', JSON.stringify(notesObj, null, 2), 'utf8', err => {
+        if (err) {
+          console.error(err);
+        }
+      });
+    });
+  } else if (keyWord === keyWordLib[3]) {
```
Comment (TimelikeClosure) [inline node-notes-cli/app.js]: Please move the names of the operations down here from the hard-coded array at the top so it's easier to tell which operation each block of code is attempting to implement.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by OSAdams** [comment]
Updated conditional expressions
(keyWord === 'read')
(keyWord === 'create')
(keyWord === 'update')
(keyWord === 'delete')

File: node-notes-cli/app.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+const keyWordLib = ['read', 'create', 'update', 'delete'];
+
+if (!keyWordLib.includes(keyWord)) {
+  console.error({ error: 'invalid command' });
+} else if (keyWordLib.includes(keyWord)) {
+  if (keyWord === 'read') {
+    fs.readFile('data.json', 'utf8', (err, data) => {
+      if (err) {
+        console.error(err);
+      }
```
Comment (TimelikeClosure) [inline node-notes-cli/app.js:13]: If there's an error reading the file, it'd make sense to stop attempting to process its contents.

File: node-notes-cli/app.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        return;
+      }
+      notesObj.notes[newId] = process.argv[3];
+      notesObj.nextId++;
+      console.log(`Note ID ${newId} successfully created`);
+      fs.writeFile('data.json', JSON.stringify(notesObj, null, 2), 'utf8', err => {
+        if (err) {
+          console.error(err);
+        }
+      });
```
Comment (TimelikeClosure) [inline node-notes-cli/app.js]: Technically, a new note isn't successfully created until its persisted to the data file, so I'd recommend waiting until that finishes successfully before letting the user know.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by OSAdams** [comment]
image

File: node-notes-cli/app.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        return;
+      }
+      notesObj.notes[newId] = process.argv[3];
+      notesObj.nextId++;
+      fs.writeFile('data.json', JSON.stringify(notesObj, null, 2), 'utf8', err => {
+        if (err) {
+          console.error(err);
+        }
+      });
+      console.log(`Note ID ${newId} successfully created`);
```
Comment (TimelikeClosure) [inline node-notes-cli/app.js]: `writeFile` is asynchronous, so this console log doesn't should before writing finishes where it is.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: node-notes-cli/app.js (hunk 4 above)
Comment (OSAdams) [reply node-notes-cli/app.js]: I overthought it. Thank you

**Comment by OSAdams** [comment]
image

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**